src/System.Linq/src/System/Linq/Take.SizeOpt.cs

```diff
@@ -3,21 +3,57 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Generic;
+using System.Diagnostics;
 
 namespace System.Linq
 {
     public static partial class Enumerable
     {
         private static IEnumerable<TSource> TakeIterator<TSource>(IEnumerable<TSource> source, int count)
         {
-            if (count > 0)
+            Debug.Assert(count > 0);
+            
+            foreach (TSource element in source)
             {
-                foreach (TSource element in source)
+                yield return element;
+                if (--count == 0) break;
+            }
+        }
+
+        private static IEnumerable<TSource> TakeLastIterator<TSource>(IEnumerable<TSource> source, int count)
+        {
+            Debug.Assert(count > 0);
+
+            if (source is IPartition<TSource> partition)
```

Comment:
None of this partition stuff should be used in the SizeOpt.  This should fail to compile in those builds.

Reply:
Was thinking about this, however found out that general Skip() method does similar checkup -> 30 in Skip.cs

Reply:
It'll actually compile; I misspoke there.  There's just nothing that'll implement it and so the condition will never be true.  So it's dead code in this .SizeOpt.cs file that shouldn't be added.  The Skip.cs file you referenced is general code used for both builds.

Reply:
Removed the redundant part.